controllers/instancegroup_controller.go

```diff
@@ -201,6 +202,10 @@ func (r *InstanceGroupReconciler) Reconcile(ctxt context.Context, req ctrl.Reque
 		return ctrl.Result{}, errors.Wrapf(err, "provisioner %v reconcile failed", provisionerKind)
 	}
 
+	if r.SetScalingConfigurationToLaunchConfig {
+		input.InstanceGroup.Spec.EKSSpec.Type = v1alpha1.LaunchConfiguration
```

Comment:
This flag is a bit strange, when set it forces specs to use launch template instead of launch configuration even though the spec requests launch config explicitly.
I think it makes more sense to introduce a flag that modifies the unspecified default.
That way resources can either omit the field and get the flag-controlled default, or explicitly declare a configuration which should be honored.

With the flag as is, you force all resources to switch and not sure it makes sense